frontend/src/components/RoundResultScreen.js

```diff
@@ -1,6 +1,43 @@
-const RoundResultsScreen = ({ players, roundResult }) => {
-  // TODO display current leader board and also indicate if the current user got the answer correct
-  return <div>Round Results Screen</div>;
+import React, { useState, useEffect } from "react";
+import "./RoundResultScreen.css";
+
+const RoundResultsScreen = ({ players, roundResult, currentUserPlayerId, timer }) => {
+    const [timeLeft, setTimeLeft] = useState(timer);
```

Comment:
remove all this code:

```
const [timeLeft, setTimeLeft] = useState(timer);
    useEffect(() => {
        setTimeLeft(timer);

        const interval = setInterval(() => {
            setTimeLeft(prevTimeLeft => Math.max(prevTimeLeft - 1, 0));
        }, 1000);

        return () => {
            clearInterval(interval);
        };
    }, [timer]);
```

the timer will be managed by the server via the `timeTick` event, which is updated via `setTimer` in the other `page.js` file

Reply:
We could let the backend handle the timer by updating `startRound` like: 

```
const startRound = (io, roomId) => {
  const game = games[roomId];
  const currentQuestion = game.questions[game.currentQuestionIndex];
  const sentQuestion = JSON.parse(JSON.stringify(currentQuestion));
  delete sentQuestion["correctAnswer"];
  io.to(roomId).emit("nextQuestion", sentQuestion);

  const roundDuration = game.gameRules.snippetLength;
  let timeLeft = roundDuration;
  const timer = setInterval(() => {
    io.to(roomId).emit("timerTick", { timeLeft });
    timeLeft--;

    if (timeLeft < 0) {
      clearInterval(timer);
      io.to(roomId).emit("roundEnded");

      let roundTransitionTimeLeft = 10;
      const roundTransitionTimer = setInterval(() => {
        io.to(roomId).emit("timerTick", { roundTransitionTimeLeft });
        roundTransitionTimeLeft--;

        if (roundTransitionTimeLeft < 0) {
          clearInterval(roundTransitionTimer);
          if (game.currentQuestionIndex < game.questions.length - 1) {
            game.currentQuestionIndex++;
            startRound(io, roomId);
          } else {
            endGame(io, roomId);
          }
        }
      }, 1000);
    }
  }, 1000);
};
```

what this does is:
- send the question, then tick for 30 seconds (or the "snippetLength")
- once we reach 0, emit a "roundEnded" event, then tick for 10 seconds
- once we reach 0, start the round if there are more questions, otherwise, end the game